src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -772,6 +775,9 @@ void ObjectCopyRequest<I>::compute_zero_ops() {
           }
 
           object_exists = (object_extent.offset > 0 || hide_parent);
+	  if (!object_exists) {
+	    update_map_nonexistent = true;
```

Comment:
I am not sure it is right. Consider a case when you have several extents, and the first extent has the offset 0. Then on the first iteration `object_exists` will be `false`, and `update_map_nonexistent` will be set to `true`, on the second iteration (an extent with the offset > 0) the `object_exists` will become `true` and `update_map_nonexistent` will remain true. Is it what you wanted?

Reply:
Yes, because the object map state is only processed after all the iterations.

Reply:
With the last change I am not sure I understand why we need `update_map_nonexistent` at all. Wouldn't it be always `!object_exists`?

Reply:
I can't remember why I did this in the first place hence removing it.

Reply:
I've added this back as the TestMigration.Stress2 fails without it. I cannot remember why exactly but will look into it again.

Reply:
Explanation:
As object_exists is usually initialized to false, it is not sufficient to check only for its value. We need to ensure that the object does not exist based processing zero_extents. Hence the new update_map_nonexistent variable.
We also need to check that object actually exists for a particular snap before attempting to update its state in the object_map as an ERANGE error will cause the object Map to invalidate both the in-memory and on-disk object map.